core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -240,13 +240,10 @@ public void commitTransaction() {
   }
 
   private void commitCreateTransaction() {
-    // fix up the snapshot log, which should not contain intermediate snapshots
-    TableMetadata createMetadata = current.removeSnapshotLogEntries(intermediateSnapshotIds);
-
     // this operation creates the table. if the commit fails, this cannot retry because another
     // process has created the same table.
     try {
-      ops.commit(null, createMetadata);
+      ops.commit(null, current);
```

Comment:
Transactions no longer need to clean up the snapshot log. That cleanup is done in `build` based on the changes since the last commit.